tutor/src/models/course/scores/task-result.js

```diff
@@ -33,11 +33,23 @@ export default class TaskResult extends BaseModel {
   @field recovered_exercise_count;
 
   @belongsTo({ model: 'course/scores/student' }) student;
+  @computed get period() { return this.student.period; }
 
   @computed get columnIndex() {
     return findIndex(this.student.data, s => s === this);
   }
 
+  @computed get progress() {
+    let count;
+    if ('homework' === this.type) {
+      count = this.completed_on_time_exercise_count + this.completed_accepted_late_exercise_count;
+      return count / this.exercise_count;
```

Comment:
I think this denominator needs to include placeholder steps. In case they did not do enough exercises to trigger the placeholders.